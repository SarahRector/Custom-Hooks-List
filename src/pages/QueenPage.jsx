import React from 'react';
import { useState } from 'react';
import QueenList from '../components/all-queens/QueenList';

const QueenPage = () => {
  const [page, setPage] = useState(1);

  return (
    <>
      <button
        onClick={() => setPage(page => page - 1)}
        disabled={page <= 1}
      >&lt;</button>

      <button
        onClick={() => setPage(page => page + 1)}
        disabled={page >= 25}
      >&gt;</button>
      <QueenList page={page} />
    </>
  );
};

export default QueenPage;
