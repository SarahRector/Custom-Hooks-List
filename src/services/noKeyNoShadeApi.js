/* eslint-disable max-len */
export const getQueens = async(page = 1) => {
  const res = await fetch(`http://www.nokeynoshade.party/api/queens/all?page=${page}`);
  const json = await res.json();

  if(!res.ok) throw 'Unable to fetch';

  return json.results.map(queen => ({
    id: queen.id,
    name: queen.name,
    image_url: queen.image
  }));
};

export const getQueensById = async(id) => {
  const res = await fetch(`http://www.nokeynoshade.party/api/queens/${id}`);
  const json = await res.json();

  if(!res.ok) throw 'Unable to fetch';

  return {
    id: json.id,
    name: json.name,
    image_url: json.image,
    quote: json.quote
  };
};
