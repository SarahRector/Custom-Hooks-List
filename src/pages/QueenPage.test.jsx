import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import QueenPage from './QueenPage';
import { MemoryRouter } from 'react-router-dom';
import { getQueens } from '../services/noKeyNoShadeApi';

jest.mock('../services/noKeyNoShadeApi.js');

describe('QueenPage component', () => {
  afterEach(() => cleanup());
  it('renders QueenPage', () => {
    getQueens.mockResolvedValue([
      { id: 89, name: 'Trixie', image_url: 'test.png' }
    ]);
    const { asFragment } = render(<MemoryRouter>
      <QueenPage />
    </MemoryRouter>);
    return waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    }); 
  });
});
