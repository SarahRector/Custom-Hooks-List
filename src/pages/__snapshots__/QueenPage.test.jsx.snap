// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueenPage component renders QueenPage 1`] = `
<DocumentFragment>
  <button
    disabled=""
  >
    &lt;
  </button>
  <button>
    &gt;
  </button>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;

exports[`QueenPage component renders QueenPage 2`] = `
<DocumentFragment>
  <button
    disabled=""
  >
    &lt;
  </button>
  <button>
    &gt;
  </button>
  <ul
    data-testid="queens"
  >
    <li>
      <a
        href="/queens/89"
      >
        <figure>
          <img
            alt="Trixie"
            src="test.png"
          />
          <figcaption>
            Trixie
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
